dags/common/enricher.py

```diff
@@ -0,0 +1,113 @@
+import os
+import xml.etree.ElementTree as ET
+from typing import Dict
+
+import backoff
+import requests
+from structlog import get_logger
+
+
+class Enricher(object):
+    def __init__(self) -> None:
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+
+    def _get_schema(self):
+        return os.getenv("REPO_URL", "http://repo.qa.scoap3.org") + "/schemas/hep.json"
+
+    @backoff.on_exception(
+        backoff.expo, requests.exceptions.RequestException, max_time=60, max_tries=3
+    )
+    def _get_arxiv_categories(self, arxiv_id=None, title=None, doi=None):
+        if arxiv_id is None and title is None and doi is None:
+            raise ValueError(
+                "One of the arxiv_id, title and doi parameters has to be different then None."
+            )
+
+        def clean_arxiv(arxiv):
+            if arxiv is None:
+                return None
+
+            return arxiv.split(":")[-1].split("v")[0].split(" ")[0].strip("\"'")
```

Comment:
could you please send me a use case of this code?

Reply:
imo we can simplify this with a simple regex `(\d{4}\.\d{4,5})` this will always get `1807.07447` the numbers between the dot

Reply:
This is a great idea ! Added 👍